internal/kinds/capp/utils/route.go

```diff
@@ -43,79 +36,56 @@ func IsDNSRecordAvailable(ctx context.Context, k8sClient client.Client, name, na
 	return available, nil
 }
 
-// GetDNSConfig returns the data of the DNS ConfigMap.
-func GetDNSConfig(ctx context.Context, k8sClient client.Client) (map[string]string, error) {
-	routeCM := corev1.ConfigMap{}
-	if err := k8sClient.Get(ctx, types.NamespacedName{Namespace: CappNS, Name: dnsCM}, &routeCM); err != nil {
-		return nil, fmt.Errorf("could not fetch configMap %q from namespace %q: %w", dnsCM, CappNS, err)
+// GetDNSConfig returns the data of the DNS for the CappConfig CRD.
+func GetDNSConfig(ctx context.Context, k8sClient client.Client) (cappv1alpha1.DNSConfig, error) {
+	cappConfig := cappv1alpha1.CappConfig{}
+	if err := k8sClient.Get(ctx, types.NamespacedName{Namespace: CappNS, Name: CappConfigName}, &cappConfig); err != nil {
+		return cappv1alpha1.DNSConfig{}, fmt.Errorf("could not fetch cappConfig %q from namespace %q: %w", CappConfigName, CappNS, err)
 	}
-
-	return routeCM.Data, nil
+	return cappConfig.Spec.DNSConfig, nil
 }
 
-// GetDNSRecordFromConfig returns the DNSRecord to be used for the record from a ConfigMap.
-func GetDNSRecordFromConfig(dnsConfig map[string]string) (string, error) {
-	dnsRecord := ""
-	var ok bool
-
-	if len(dnsConfig) > 0 {
-		dnsRecord, ok = dnsConfig[cnameKey]
-		if !ok {
-			return dnsRecord, fmt.Errorf("%q key is not set in ConfigMap %q", cnameKey, dnsCM)
-		} else if dnsRecord == "" {
-			return dnsRecord, fmt.Errorf("%q is empty in ConfigMap %q", cnameKey, dnsCM)
-		}
-	}
+// GetDNSRecordFromConfig returns the DNSRecord to be used for the record from a CappConfig CRD.
+func GetDNSRecordFromConfig(dnsConfig cappv1alpha1.DNSConfig) (string, error) {
+	dnsRecord := dnsConfig.CNAME
 
+	if dnsRecord == "" {
+		return "", fmt.Errorf("%q is empty in CappConfig DNSConfig data", "CNAME")
+	}
 	return dnsRecord, nil
 }
 
-// GetZoneFromConfig returns the zone to be used for the record from a ConfigMap.
-func GetZoneFromConfig(dnsConfig map[string]string) (string, error) {
-	var ok bool
-	zone := placeholderZone
-	if len(dnsConfig) > 0 {
-		zone, ok = dnsConfig[zoneKey]
-		if !ok {
-			return zone, fmt.Errorf("%q key is not set in ConfigMap %q", zoneKey, dnsCM)
-		} else if zone == "" {
-			return zone, fmt.Errorf("%q is empty in ConfigMap %q", zoneKey, dnsCM)
-		} else if !strings.HasSuffix(zone, dot) {
-			return zone, fmt.Errorf("%q value must end with a %q in ConfigMap %q", zoneKey, dot, dnsCM)
-		}
+// GetZoneFromConfig returns the zone to be used for the record from the CappConfig CRD.
+func GetZoneFromConfig(dnsConfig cappv1alpha1.DNSConfig) (string, error) {
```

Comment:
Not in the scope for this PR, but create an issue to rework these functions to use generics.
Currently they all do pretty much the same thing